### PR TITLE
Adding Travis check for PR code only

### LINKED ISSUE #1832: Implement linting test for PR files in Travis
Use the solution proposed here: https://medium.com/@joey_9999/how-to-only-lint-files-a-git-pull-request-modifies-3f02254ec5e0 and integrate into Travis. The goal is to be able to test linting only of the files of a PR, not of the whole package.

### PR DESCRIPTION
**Pull request by iparask**
I'm changing the purpose of the PR. This one closes #1832.

It fails because it picks up the changes in the `yml` file as well. Having said that it takes the files changed by this PR and executes `flake8` and `pylint` only on those.

### CODE REVIEW

**Comment by itomaldonado** [comment]
Can one of the admins verify this patch?

File: .travis.yml [hunk 1]
```diff
@@ -15,8 +15,8 @@ git:
 
 env:
   global:
-    - BRANCH=`git log --format="%H" -n 1`
-    - DIFF=`git diff --name-only --diff-filter=b HEAD...$BRANCH`
+    - LOG=`git log -n 1 | grep Merge`
+    - DIFF=`git diff --name-only --diff-filter=b ${LOG:7:8}...${LOG:16:8}`
```
Comment (mturilli) [inline .travis.yml]: I am a bit worried this is fragile. $LOG looks like: `Merge: 5cb84ac4 5de8828b` so would something like this work better?
```
OLD=`echo $LOG | cut -d'  ' -f2`
NEW=`echo $LOG | cut -d'  ' -f3`
```

Comment (iparask) [reply .travis.yml]: Thank you for the suggestion

**Comment by iparask** [comment]
@andre-merzky feel free to merge whenever you want!

**Comment by andre-merzky** [comment]
Thanks!

**Pull request merged**